Week8/woojo230/mission01/src/pages/search.tsx

```diff
@@ -0,0 +1,129 @@
+import { useState, useCallback, useEffect, useRef } from 'react';
+import Error from '../components/error';
+import LoadingSpinner from '../components/loading';
+import useInfiniteScrollObserver from '../hooks/observer/useInfiniteScrollObserver';
+import useGetInfiniteScroll from '../hooks/queries/infiniteScroll/useGetInfiniteScroll';
+import { LpGrid } from '../components/lp/lpGrid';
+import { OrderButtons } from '../components/lp/orderBtn';
+import { useDebounce } from '../hooks/useDebounce';
+import useGetTagInfiniteScroll from '../hooks/queries/infiniteScroll/useGetTagInfinityScroll';
+import SearchForm from '../components/search/searchForm';
+import TagBtn from '../components/search/tagBtn';
+
+type SearchType = 'title' | 'tag';
+
+const Search = () => {
+  const [search, setSearch] = useState('');
+  const [order, setOrder] = useState('desc');
+  const [searchType, setSearchType] = useState<SearchType>('title');
+  const debouncedSearch = useDebounce(search, 500);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const {
+    data: lpData,
+    fetchNextPage,
+    hasNextPage,
+    isFetching,
+    isLoading,
+    error,
+    refetch,
+  } = useGetInfiniteScroll({
+    order: order,
+    limit: 50,
+    search: searchType === 'title' ? debouncedSearch : '',
+  });
+
+  const {
+    data: tagData,
+    fetchNextPage: tagFetchNextPage,
+    hasNextPage: tagHasNextPage,
+    isFetching: tagIsFetching,
+    isLoading: tagIsLoading,
+    error: tagError,
+    refetch: tagRefetch,
+  } = useGetTagInfiniteScroll({
+    tagName: searchType === 'tag' ? debouncedSearch : '',
+    order: order,
+    limit: 50,
+    search: '',
+    enabled: searchType === 'tag' && debouncedSearch !== '',
+  });
+
+  const allLpData = lpData?.pages.flatMap((page) => page.data) ?? [];
+  const allTagData = tagData?.pages.flatMap((page) => page.data) ?? [];
+
+  const observerRef = useInfiniteScrollObserver(
+    searchType === 'title' ? hasNextPage : tagHasNextPage,
+    searchType === 'title' ? isFetching : tagIsFetching,
+    searchType === 'title' ? fetchNextPage : tagFetchNextPage
+  );
+
+  const handleClick = () => {
+    if (searchType === 'title') {
+      setSearchType('tag');
+      setSearch('');
+    } else {
+      setSearchType('title');
+      setSearch('');
+    }
+  };
```

Comment:
이 부분도 각 조건의 로직이 비슷하므로, 다른 부분처럼 삼항 연산자를 사용하면 더 간결하고 가독성이 향상될 것 같습니다!
```suggestion
  const handleClick = () => {
    setSearchType(prevType => prevType === "title" ? "tag" : "title");
    setSearch("");
};
```

Reply:
수정했습니당